### PR TITLE
Add definitions for Xmake as an addon

### PR DESCRIPTION
**Pull request by LelouchHe**
This pr is for #24

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -63,4 +63,8 @@
 	url = https://github.com/LuaCATS/tex-luatex.git
 [submodule "addons/slnunicode/module"]
 	path = addons/slnunicode/module
-	url = https://github.com/LuaCATS/slnunicode.git
\ No newline at end of file
+	url = https://github.com/LuaCATS/slnunicode.git
+[submodule ".\\addons\\xmake\\module"]
+	path = .\\addons\\xmake\\module
```
Comment (carsakiller) [inline .gitmodules:68]: Please use UNIX paths
```suggestion
[submodule "addons/xmake/module"]
	path = addons/xmake/module
```

File: .gitmodules [hunk 2]
```diff
@@ -63,4 +63,8 @@
...
+	url = https://github.com/LelouchHe/xmake-luals-addon
```
Comment (carsakiller) [inline .gitmodules:69]: This is not a valid git clone URL. Please use the HTTPS clone URL provided by GitHub. It should look like `https://github.com/USER/REPO.git`

**Review by carsakiller — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (LelouchHe) [reply .gitmodules:68]: sure. This content is generated by running `git submodule`. let me update it manually

File: .gitmodules (hunk 2 above)
Comment (LelouchHe) [reply .gitmodules:69]: sure. this is generated by running `git submodule` cmd. let me update it manually

**Comment by LelouchHe** [comment]
it seems that when resolving conflicts, i removed the tex-lualatex. trying to revert that

**Comment by LelouchHe** [comment]
the revert will make the history looks bad. let me create a new one instead.

**Comment by carsakiller** [comment]
You can undo the previous commits and force push the new history

**Comment by LelouchHe** [comment]
> You can undo the previous commits and force push the new history

i have created a new one already. #28

**Pull request closed without merging**